core/Network/TLS/Types.hs

```diff
@@ -19,6 +19,8 @@ module Network.TLS.Types
 import Data.ByteString (ByteString)
 import Data.Word
 
+type HostName = String
```

Comment:
the goal of this alias was to remember to do something sensible about this, not really expose it to the user. A real `Hostname` type is coming soon (couple of months away), and when that happens, I'm planning to move those types in tls and x509 to use it.